pkg/aspect/config/attrgen.proto

```diff
@@ -0,0 +1,52 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package pkg.aspect.config;
+
+import "gogoproto/gogo.proto";
+
+option go_package="config";
+option (gogoproto.goproto_getters_all) = false;
+option (gogoproto.equal_all) = false;
+option (gogoproto.gostring_all) = false;
+
+// Configures an AttributesGenerator aspect.
+message AttributesGeneratorParams {
```

Comment:
One adapter may have a completely different set of inputs from a second adapter that _also_ implements the `attributes` aspect. We can't use an actual message for inputs because they will be different. The same is true for outputs.

At the _adapter_ level, an adapter takes a map[string]interface{} and produces a map[string]interface{}. The manager, using _aspect_ config translates in both directions.

The example in the PR description provides an example. A mixer version adapter and a k8s pod info adapter take different inputs and produce different outputs.